processor/src/main/java/com/linecorp/decaton/processor/runtime/ProcessPipeline.java

```diff
@@ -60,28 +60,49 @@ public ProcessPipeline(ThreadScope scope,
         processMetrics = metrics.new ProcessMetrics();
     }
 
-    public CompletableFuture<Void> scheduleThenProcess(TaskRequest request) throws InterruptedException {
-        DecatonTask<T> extracted = extract(request);
+    public void scheduleThenProcess(TaskRequest request) throws InterruptedException {
+        final DecatonTask<T> extracted;
+        try {
+            extracted = extract(request);
+        } catch (RuntimeException e) {
+            log.error("Dropping not-deserializable task [{}]", request.id(), e);
+            taskMetrics.tasksDiscarded.increment();
+            return;
```

Comment:
Isn't it necessary to call `completion.complete()` ?
IIUC Decaton shouldn't stop processing due to invalid tasks

https://github.com/line/decaton/blob/master/docs/consuming-any-data.adoc
![image](https://user-images.githubusercontent.com/6070684/78517331-c7be8200-77f7-11ea-91c2-3eef45255f77.png)

Reply:
doh! super nice catch, thanks so much!
I'll add a test case to cover it and fix it to handle completion.